pkgs/development/libraries/epoll-shim/default.nix

```diff
@@ -0,0 +1,38 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "epoll-shim";
+  version = "unstable-2023-02-05";
+
+  src = fetchFromGitHub {
+    owner = "jiixyj";
+    repo = finalAttrs.pname;
+    rev = "702e845d7850e30a7b9e29f759c9c8f7bb40784b";
+    hash = "sha256-QfBnF0/P2KjQggEdJCdqVJDeV/+iaN0OZIwIGyIyr68=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+  ];
+
+  cmakeFlags = [
+    "-DCMAKE_INSTALL_PKGCONFIGDIR=${placeholder "out"}/lib/pkgconfig"
+    "-DBUILD_TESTING=${lib.boolToString finalAttrs.doCheck}"
+  ];
```

Comment:
Ah right I remember, to cross compile it needs:

```suggestion
  ] ++ lib.optional (stdenv.hostPlatform != stdenv.buildPlatform)
    "-DALLOWS_ONESHOT_TIMERS_WITH_TIMEOUT_ZERO=${lib.boolToString {
      dragonfly = true;
      freebsd = true;
      netbsd = false;
      openbsd = true;
    }.${stdenv.hostPlatform.parsed.kernel.name}}";
```

I don't know what the right value for Darwin is though — you'd have to check in the CMake cache.

(And if you'd rather skip this and let me fix it in a follow up PR, that's fine.)

Reply:
Feel free to update my PR or create a new one (and add yourself as a maintainer).

Reply:
> I've got `ALLOWS_ONESHOT_TIMERS_WITH_TIMEOUT_ZERO` evaluated to true on my builds, but they are clearly not cross compiled, so I can't say much.

That's what I was looking for! When natively compiling, it can run a test program, but when cross compiling, it can't do that, so we have to check in advance what value it produces for native compilation on each platform.